### PR TITLE
Party like it s 2016

### PR DESCRIPTION
**Pull request by fallwith**
Ensure that all Ruby files instruct Ruby to freeze string literals by
default

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -1361,9 +1361,6 @@ Style/FormatString:
 Style/FormatStringToken:
   Enabled: false
 
-Style/FrozenStringLiteralComment:
-  Enabled: false
-
```
Comment (fallwith) [inline .rubocop.yml:1366]: `frozen_string_literal` was introduced on 2015-12-15 and the RuboCop style cop to enforce its usage with a `true` value was introduced soon after in early 2016. By simply no longer disabling the cop, we'll benefit from the default behavior of it enforcing the use of `frozen_string_literal: true`. Party like it's 2016!

File: test/new_relic/agent/database_test.rb [hunk 2]
```diff
@@ -182,9 +183,9 @@ def test_explain_sql_one_select_with_pg_explain_string_result
     config = {:adapter => 'postgresql'}
     sql = 'select count(id) from blogs limit 1'
 
-    plan = "Limit  (cost=11.75..11.76 rows=1 width=4)
+    plan = String.new("Limit  (cost=11.75..11.76 rows=1 width=4)
```
Comment (fallwith) [inline test/new_relic/agent/database_test.rb:186]: This string gets mutated, needs `String.new`

File: test/new_relic/agent/database_test.rb [hunk 3]
```diff
@@ -453,7 +454,7 @@ def test_capture_query_long_query
     assert_equal('a' * (NewRelic::Agent::Database::MAX_QUERY_LENGTH - 3) + '...', truncated_query)
   end
 
-  INVALID_UTF8_STRING = (''.respond_to?(:force_encoding) ? "select \x80".force_encoding('UTF-8') : "select \x80")
+  INVALID_UTF8_STRING = String.new("select \x80").force_encoding('UTF-8')
```
Comment (fallwith) [inline test/new_relic/agent/database_test.rb:457]: This string gets mutated, needs `String.new`. NOTE: we no longer test with rubies that don't respond to `#force_encoding`.

**Review by kaylareopelle — APPROVED** [review]
Thanks for taking care of this, James!

**Pull request merged**